src/pages/magic-link/magic-link.ts

```diff
@@ -48,73 +50,60 @@ export class MagicLinkPage {
-      this.cacheService.setLocal('teams', data.Teams);
-      // get game_id data after login
-      this.gameService.getGames()
-          .subscribe(
-            data => {
-              console.log("game data: ", data);
-              _.map(data, (element) => {
-                console.log("game id: ", element[0].id);
-                this.cacheService.setLocal('game_id', element[0].id);
-              });
-            },
-            err => {
-              console.log("game err: ", err);
-            }
-          );
-      // get milestone data after login
-      this.authService.getUser()
-        .subscribe(
-          data => {
-            this.cacheService.setLocal('name', data.User.name);
-            this.cacheService.setLocal('email', data.User.email);
-            this.cacheService.setLocal('program_id', data.User.program_id);
-            this.cacheService.setLocal('project_id', data.User.project_id);
-          },
-          err => {
-            console.log(err);
-          }
-        );
-      // get milestone data after login
-      this.milestoneService.getMilestones()
-        .subscribe(
-          data => {
-            console.log(data[0].id);
-            this.milestone_id = data[0].id;
-            this.cacheService.setLocal('milestone_id', data[0].id);
-            console.log("milestone id: " + data[0].id);
-            loading.dismiss();
-            this.navCtrl.push(TabsPage).then(() => {
-              window.history.replaceState({}, '', window.location.origin);
-            });
-          },
-          err => {
-            console.log(err);
-          }
-        )
-      this.cacheService.write('isAuthenticated', true);
-      this.cacheService.setLocal('isAuthenticated', true);
-      },
-      err => {
-      const failAlert = this.alertCtrl.create({
-        title: 'Magic did NOT happen',
-        message: this.misMatchTokenErrMessage,
-        buttons: ['Close']
-      });
-      failAlert.present();
-        this.navCtrl.push(LoginPage).then(() => {
-          window.history.replaceState({}, '', window.location.origin);
-        });
-        // console.log("Login failed");
-        this.cacheService.removeLocal('isAuthenticated');
-        this.cacheService.write('isAuthenticated', false);
-      });
+    loading.present().then(() => {
+      observable.subscribe(
+        data => {
+          this.cacheService.setLocal('apikey', data.apikey);
+          this.cacheService.setLocal('timelineID', data.Timelines[0].Timeline.id);
+          this.cacheService.setLocal('teams', data.Teams);
+          let getGame = this.gameService.getGames();
+          let getUser = this.authService.getUser();
+          let getMilestone = this.milestoneService.getMilestones();
+          Observable.forkJoin([getGame, getUser, getMilestone])
+            .subscribe(
+              results => {
+                // results[0] game API data
+                this.gameID = results[0].Games[0].id;
```

Comment:
@DamengRandom here has the same logic code as in *login component* above. If same logic code is required for different component, it would be better to write an additional function and reuse the same code.